pkg/cluster/mpi/statefulset.go

```diff
@@ -118,14 +118,29 @@ func (c statefulSetComponent) Reconcile(ctx *core.Context) (ctrl.Result, error)
 
 func (c statefulSetComponent) Finalize(ctx *core.Context) (ctrl.Result, bool, error) {
 	cr := objToMPICluster(ctx.Object)
+
 	pvcListOpts := []client.ListOption{
 		client.InNamespace(cr.Namespace),
 		client.MatchingLabels(meta.MatchLabels(cr)),
 	}
-
 	err := actions.DeleteStorage(ctx, pvcListOpts)
+	if err != nil {
+		return ctrl.Result{}, false, fmt.Errorf("cannot delete storage: %w", err)
```

Comment:
So if this fails, does the CR just stay stuck deleting? Will it retry at some point?

Reply:
I suppose so. It's a valid question, but it's out of scope of this specific piece of code. I mean, this function obviously expects an error to be returned if something happened. Then however calls it (Sonny's custom reconciler) is supposed to do something reasonable. I believe it does, but I have never tried this sort of things explicitly. 
I will verify what's going on.

Reply:
I am going to change this to explicitly tell it to retry.